src/components/Trashy/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+import TrashySchedule from 'components/Trashy/TrashySchedule';
+import TrashyAddress from 'components/Trashy/TrashyAddress';
+
+const nextBulkPickupDate = new Date().toString();
+
+class Trashy extends Component {
+  state = {
+    address: null,
+    addressSuggestions: [],
+    pickupDates: [],
+    nextBulkPickupDate: null,
+  };
+
+  getAddressSuggestions = addressText => {
+    axios
+      .post('https://recollect-wrapper.herokuapp.com/api/suggest', {
+        address: addressText,
+      })
+      .then(response => this.setState({ addressSuggestions: response.data }))
+      .catch(error => console.log(error));
+  };
+
+  getCalendar = addressId => {
+    axios
+      .post('https://recollect-wrapper.herokuapp.com/api/calendar', {
+        id: addressId,
+      })
+      .then(response => this.setState({ pickupDates: response.data }))
+      .catch(error => console.log(error));
+  };
+
+  getNextBulkPickupDate = addressId => {
+    axios
+      .post('https://recollect-wrapper.herokuapp.com/api/nextbulkpickup', {
+        id: addressId,
+      })
+      .then(response => this.setState({ nextBulkPickupDate: response.data[0] }))
+      .catch(error => console.log(error));
+  };
+
+  getAddressId = address => {
+    axios
+      .post('https://recollect-wrapper.herokuapp.com/api/address', address)
+      .then(response => response.data)
+      .then(addressId => {
+        this.getCalendar(addressId);
+        this.getNextBulkPickupDate(addressId);
+      })
+      .catch(error => console.log(error));
+  };
+
+  setAddress = address => {
+    this.setState({ address: address });
+    this.getAddressId(address);
+  };
+
+  render() {
+    const { intl, options } = this.props;
+    return (
+      <div className="coa-Trashy">
+        <TrashyAddress
+          suggestions={this.state.addressSuggestions}
+          setAddress={this.setAddress}
+          getSuggestions={this.getAddressSuggestions}
+        />
+        {this.state.address && (
+          <TrashySchedule
```

Comment:
Also curious why TrashySchedule doesn't manage fetching pickupDates and nextBulkPickupDate on its own too. It doesn't seem like other components need awareness of this data or process. Looks like TrashySchedule only needs an addressId and on componentWillReceiveProps within TrashySchedule you'd trigger a fetch? Again, It's very likely that I'm overlooking something or that there's a better reason to structure React components this way.

Reply:
As we were building it out we didn't know which calls would give us what data, so we decided to try to abstract as much of the logic around fetching data from the api out of any of the display components, this also made it much easier to develop the schedule within storybook.

I'm not married to this implementation, and it's definitely a discussion I'm interested in having as we move towards a more robust implementation of the recollect api: https://github.com/cityofaustin/techstack/issues/444